typed-racket-lib/typed-racket/static-contracts/constraints.rkt

```diff
@@ -204,7 +203,9 @@
     [(list (contract-restrict vs rec constraints) ...)
      (contract-restrict (merge-kind-maxes min vs)
                         (free-id-table-union rec)
-                        (apply set-union (set) constraints))]))
+                        (for/fold ([h (set)])
```

Comment:
it is not (I wrote it that way when comparing sets versus hash tables and forgot about it) -- I'll revert that before pushing.